library/ecdh.c

```diff
@@ -284,7 +286,9 @@ int mbedtls_ecdh_make_public( mbedtls_ecdh_context *ctx, size_t *olen,
                       void *p_rng )
 {
     int ret;
+#if !defined(MBEDTLS_ECDH_GEN_PUBLIC_ALT)
```

Comment:
We can't do that, because now compilation will fail if `MBEDTLS_ECP_RESTARTABLE` and `MBEDTLS_ECDH_GEN_PUBLIC_ALT` are defined simultaenously.

Reply:
Why? This was a fix for compilation warning.
In case `MBEDTLS_ECDH_GEN_PUBLIC_ALT ` is defined, the local `rs_ctx` variable will not be used, because of using the non restartable version of `mbedtls_ecdh_gen_public()`, in line [302](https://github.com/ARMmbed/mbedtls/pull/2165/files#diff-97f71deba621674b90985642f0884ebdR302)

Reply:
See L.294